src/QMCHamiltonians/SOECPotential.h

```diff
@@ -17,55 +17,102 @@
                                        Vector<ValueType>& dhpsioverpsi) override;
 
+  void mw_evaluate(const RefVectorWithLeader<OperatorBase>& o_list,
+                   const RefVectorWithLeader<TrialWaveFunction>& wf_list,
+                   const RefVectorWithLeader<ParticleSet>& p_list) const override;
+
+  void mw_evaluatePerParticle(const RefVectorWithLeader<OperatorBase>& o_list,
+                              const RefVectorWithLeader<TrialWaveFunction>& wf_list,
+                              const RefVectorWithLeader<ParticleSet>& p_list,
+                              const std::vector<ListenerVector<Real>>& listeners,
+                              const std::vector<ListenerVector<Real>>& listeners_ions) const override;
+
   bool put(xmlNodePtr cur) override { return true; }
 
   bool get(std::ostream& os) const override
   {
-    os << "SOECPotential: " << IonConfig.getName();
+    os << "SOECPotential: " << IonConfig_.getName();
     return true;
   }
 
   std::unique_ptr<OperatorBase> makeClone(ParticleSet& qp, TrialWaveFunction& psi) final;
 
   void addComponent(int groupID, std::unique_ptr<SOECPComponent>&& pp);
 
-  void setRandomGenerator(RandomGenerator* rng) override { myRNG = rng; }
+  void setRandomGenerator(RandomGenerator* rng) override { myRNG_ = rng; }
+
+  //initialize shared resource and hand to collection
+  void createResource(ResourceCollection& collection) const override;
+
+  //acquire a shared resource
+  void acquireResource(ResourceCollection& collection, const RefVectorWithLeader<OperatorBase>& o_list) const override;
+
+  //return a shared resource to a collection
+  void releaseResource(ResourceCollection& collection, const RefVectorWithLeader<OperatorBase>& o_list) const override;
 
 protected:
-  RandomGenerator* myRNG;
-  std::vector<SOECPComponent*> PP;
-  std::vector<std::unique_ptr<SOECPComponent>> PPset;
-  ParticleSet& IonConfig;
-  TrialWaveFunction& Psi;
+  RandomGenerator* myRNG_;
+  std::vector<SOECPComponent*> PP_;
+  std::vector<std::unique_ptr<SOECPComponent>> PPset_;
+  ParticleSet& IonConfig_;
+  TrialWaveFunction& Psi_;
+  static void mw_evaluateImpl(const RefVectorWithLeader<OperatorBase>& o_list,
+                              const RefVectorWithLeader<TrialWaveFunction>& wf_list,
+                              const RefVectorWithLeader<ParticleSet>& p_list,
+                              std::optional<ListenerOption<Real>> listeners,
+                              bool keep_grid = false);
 
 private:
   ///number of ions
-  int NumIons;
+  int NumIons_;
   ///index of distance table for ion-el pair
-  int myTableIndex;
+  int myTableIndex_;
   ///reference to the electrons
-  ParticleSet& Peln;
+  ParticleSet& Peln_;
   ///neighborlist of electrons
-  NeighborLists ElecNeighborIons;
+  NeighborLists ElecNeighborIons_;
   ///neighborlist of ions
-  NeighborLists IonNeighborElecs;
+  NeighborLists IonNeighborElecs_;
```

Comment:
Please put all the variable names in lower case. IonNeighborElecs_-> ion_neighbor_elecs_ to be compliant with up-to-date naming convention.

Reply:
done